client/packages/inventory/src/Stocktake/context/stocktakeLineError.tsx

```diff
@@ -0,0 +1,63 @@
+import { PropsWithChildrenOnly, RecordWithId } from '@common/types';
+import React, { createContext, useContext, useState } from 'react';
+import {
+  AdjustmentReasonNotProvidedErrorFragment,
+  AdjustmentReasonNotValidErrorFragment,
+  SnapshotCountCurrentCountMismatchErrorFragment,
+  StockLineReducedBelowZeroErrorFragment,
+} from '../api';
+
+export type StocktakeLineError =
+  | AdjustmentReasonNotProvidedErrorFragment
+  | StockLineReducedBelowZeroErrorFragment
+  | AdjustmentReasonNotValidErrorFragment
+  | SnapshotCountCurrentCountMismatchErrorFragment;
+
+const useStocktakeLineErrors = () => {
+  const [errors, setErrors] = useState<{
```

Comment:
if this approach is used in another area, much of the context would be repeated.. I could see a lot of it being made generic - just passing in the `StocktakeLineError` type and using that throughout. I think not now.. we can do that as needed

Reply:
Yeah I agree ! We could have generic methods that we use to construct typed context (otherwise we loose the typing)